### PR TITLE
Frontend Preact X

### PR DESCRIPTION
**Pull request by thisconnect**
- upgraded Preact 8 to X
- switched to popular [enzyme](https://www.npmjs.com/package/enzyme) testing utility
- refactored status.tsx
- updated i18next
- rewrote language.tsx with typescript 
- typescript fixes for Preact X
- fixed a bug in wait-dialog.tsx due to Preact children not always returning an arrays anymore

Note this changes a lot of the frontend dependencies, when testing this branch use `make buildweb` once to install all deps.

### CODE REVIEW

**Comment by thisconnect** [comment]
BB01 confirm dialog flashes on bitcoin tx success dialog

File: frontends/web/src/.babelrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "presets": [ 
+        "preact-cli/babel"
+    ]
+}
```
Comment (thisconnect) [inline frontends/web/src/.babelrc:5]: was added here https://github.com/preactjs-templates/typescript/blob/master/template/src/.babelrc

File: frontends/web/src/components/backups/backup.tsx [hunk 2]
```diff
@@ -36,21 +36,21 @@ export interface Backup {
 
 type Props = BackupsListItemProps & TranslateProps;
 
+// eslint-disable-next-line react/prefer-stateless-function
```
Comment (thisconnect) [inline frontends/web/src/components/backups/backup.tsx:39]: Keeping as TODO (change from class to pure function)

File: frontends/web/src/components/backups/backup.tsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ class BackupsListItem extends Component<Props> {
             </Radio> :
             <div>
                 <div className="text-medium m-bottom-quarter">{backup.name}</div>
-                <div className={style.backupID}>ID: {backup.id}</div>
+                <div className={style.backupId}>ID: {backup.id}</div>
```
Comment (thisconnect) [inline frontends/web/src/components/backups/backup.tsx:73]: [typed-css-modules](https://www.npmjs.com/package/typed-css-modules) changed from v0.3.x -> 0.6.6

File: frontends/web/src/components/backups/backups.tsx [hunk 4]
```diff
@@ -46,7 +47,7 @@ interface State {
 }
 
 class Backups extends Component<Props, State> {
-    private scrollableContainer!: HTMLElement;
+    private scrollableContainer: HTMLDivElement | null = null;
```
Comment (thisconnect) [inline frontends/web/src/components/backups/backups.tsx:50]: 🤔

File: frontends/web/src/components/backups/create.jsx [hunk 5]
```diff
@@ -90,7 +90,6 @@ export default class Create extends Component {
                             <form onSubmit={this.create}>
                                 <Input
                                     autoFocus
-                                    autoComplete="off"
```
Comment (thisconnect) [inline frontends/web/src/components/backups/create.jsx:93]: ts-ignore this ?

File: frontends/web/src/components/mobiledatawarning.tsx [hunk 6]
```diff
@@ -25,18 +26,15 @@ interface LoadedProps {
...
-    if (!usingMobileData) {
-        return null;
-    }
+function MobileDataWarning({ usingMobileData, t }: RenderableProps<Props>) {
     return (
-        <Status dismissable="mobile-data-warning" type="warning">
+        <Status dismissable="mobile-data-warning" type="warning" hidden={!usingMobileData}>
             {t('mobile.usingMobileDataWarning')}
         </Status>
     );
```
Comment (thisconnect) [inline frontends/web/src/components/mobiledatawarning.tsx:34]: It is not necessary (anymore?) to return `JSX.Element | null` of a pure function as long as Typescript can understand it.

File: frontends/web/src/components/transactions/transaction.tsx [hunk 7]
```diff
@@ -231,7 +231,7 @@ class Transaction extends Component<Props, State> {
                                 <label for="note">{t('note.title')}</label>
                                 <Input
                                     align="right"
-                                    autoFocus={!editMode ? 'false' : 'true'}
+                                    autoFocus={!!editMode}
```
Comment (thisconnect) [inline frontends/web/src/components/transactions/transaction.tsx:234]: This needs to be tested in real Qt app

File: frontends/web/src/declaration.d.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+declare module "*.css" {
+    const mapping: Record<string, string>;
+    export default mapping;
+}
```
Comment (thisconnect) [inline frontends/web/src/declaration.d.ts:4]: Prevents an error when importing a .css file that has no .css.d.ts file. The linter will still error with

"Property 'prettyButton' does not exist on type 'typeof import("*.css")'.ts(2339)"

File: frontends/web/src/decorators/translate.ts [hunk 9]
```diff
@@ -52,7 +51,7 @@ export type Translate = (i18nKey: string, values?: TranslateOptions) => any;
  */
 export interface TranslateProps {
     t: Translate;
-    i18n: any;
+    i18n: originalTranslate;
```
Comment (thisconnect) [inline frontends/web/src/decorators/translate.ts:54]: nice

File: frontends/web/tests/components/message/__snapshots__/message.test.tsx.snap [hunk 10]
```diff
@@ -1,13 +1,11 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`components/message/message should match the snapshot 1`] = `
-preact-render-spy (1 nodes)
--------
 <div
-  class="success"
```
Comment (thisconnect) [inline frontends/web/tests/components/message/__snapshots__/message.test.tsx.snap:7]: 🤔

File: frontends/web/tests/routes/account/send/__snapshots__/feetargets.test.tsx.snap [hunk 11]
```diff
@@ -1,57 +1,55 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`routes/account/send/feetargets should match the snapshot 1`] = `
-preact-render-spy (1 nodes)
--------
 <div
-  class="input align-left  isTransparent"
+  className="   "
```
Comment (thisconnect) [inline frontends/web/tests/routes/account/send/__snapshots__/feetargets.test.tsx.snap:5]: need to investigate

**Comment by thisconnect** [comment]
> BB01 confirm dialog flashes on bitcoin tx success dialog

```diff
diff --git a/frontends/web/src/components/wait-dialog/wait-dialog.tsx b/frontends/web/src/components/wait-dialog/wait-dialog.tsx
index 00e20c3b..aa4a629d 100644
--- a/frontends/web/src/components/wait-dialog/wait-dialog.tsx
+++ b/frontends/web/src/components/wait-dialog/wait-dialog.tsx
@@ -99,7 +99,7 @@ class WaitDialog extends Component<Props, State> {
         prequel,
         title,
         paired = false,
-        touchConfirm = true,
+        touchConfirm,
         children,
     }: RenderableProps<Props>,
     {}: State) {
```

This is the fix for the BB01 dialog flash, not sure why it doesn't overwrite the default assignment  with false have to check.

Also ok if I just squash this into the first commit?

File: frontends/web/tests/routes/account/send/__snapshots__/feetargets.test.tsx.snap (hunk 11 above)
Comment (thisconnect) [reply frontends/web/tests/routes/account/send/__snapshots__/feetargets.test.tsx.snap:5]: I can't get typed-css-modules to work in enzyme, but if we need to test those classnames we can `expect(context.find('.input').hasClass('align-left')).toBe(true);`

File: frontends/web/package.json [hunk 12]
```diff
@@ -4,43 +4,92 @@
...
+    "preset": "jest-preset-preact",
+    "setupFiles": [
+      "<rootDir>/tests/__mocks__/browserMocks.ts",
+      "<rootDir>/tests/__mocks__/setupTests.ts"
+    ],
+    "snapshotSerializers": [
+      "enzyme-to-json/serializer"
+    ],
+    "setupFilesAfterEnv": [
+      "./tests/setup.js"
```
Comment (thisconnect) [inline frontends/web/package.json:92]: double check if this is necessary

File: frontends/web/tests/setup.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+import { createSerializer } from 'enzyme-to-json';
+ 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
```
Comment (thisconnect) [inline frontends/web/tests/setup.js:5]: double check if this is necessary in addition to to snapshotSerializers in package.json
```
"snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
```

**Comment by thisconnect** [comment]
Also remove `tsconfig.esling.json` `tsconfig.test.json` and `tslint.json`

File: frontends/web/package.json [hunk 14]
```diff
@@ -4,43 +4,92 @@
   "version": "0.0.0",
   "scripts": {
     "gen-ts-css": "tcm src --camelCase dashes",
```
Comment (thisconnect) [inline frontends/web/package.json:6]: preact typescript template doesn't generate `.css.d.ts` files
https://github.com/preactjs-templates/typescript/tree/master/template

File: frontends/web/src/i18n/i18n.ts [hunk 15]
```diff
@@ -79,7 +80,6 @@ i18Init.init({
         enabled: i18nEditorActive,
         autoOpen: true,
         mode: 'iframe', // 'window',
-        projectId: locizeProjectID,
```
Comment (thisconnect) [inline frontends/web/src/i18n/i18n.ts:82]: 🤔

**Comment by thisconnect** [comment]
PTAL Sorry it's so much.

**Comment by thisconnect** [comment]
BB01 unlock screen doesn't load…

File: frontends/web/src/i18n/i18n.ts (hunk 15 above)
Comment (thisconnect) [reply frontends/web/src/i18n/i18n.ts:82]: I haven't tested the special build with the in-app locize editor, had to up the locize npm module typescript now complains about projectId, I am affraid that with this upgrade the in app editor can't work anymore.

**Comment by x1ddos** [comment]
hey @thisconnect, how's react coming? :)

**Comment by thisconnect** [comment]
> hey @thisconnect, how's react coming? :)

@x1ddos extracted some findings in this PR into https://github.com/digitalbitbox/bitbox-wallet-app/pull/1246 this is prerequisite for react. going to care about multiaccount ui in the meantime.

**Comment by thisconnect** [comment]
closing in favor of https://github.com/digitalbitbox/bitbox-wallet-app/pull/1531

**Pull request closed without merging**